### PR TITLE
Relax the naming rule of extension name

### PR DESCRIPTION
**Pull request by kersten1**
Update from this PR - https://github.com/riscv/riscv-isa-manual/pull/718

Not convinced that I did the supervisor section correctly.  But I'm also not convinced that it is explained in Volume II. However, volume II has grown significantly and I just do not know it all.

### CODE REVIEW

File: src/naming.adoc [hunk 1]
```diff
@@ -76,11 +76,11 @@ of the P extension.
...
 === Additional Standard Unprivileged Extension Names
 
-Standard unprivileged extensions can also be named using a single "Z" followed by
-an alphabetical name and an optional version number. For example,
-"Zifencei" names the instruction-fetch fence extension described in
-<<zifencei>>; "Zifencei2" and
-"Zifencei2p0" name version 2.0 of same.
+Standard unprivileged extensions can also be named by using a single "Z" followed by an
+alphanumeric name. The name must end with an alphabetical character or an 
+optional version number. The second letter from the bottom cannot be a numeric if the
```
Comment (aswaterman) [inline src/naming.adoc:81]: "a numeric" => "numeric"

Comment (aswaterman) [reply src/naming.adoc:81]: "bottom" => "end"

File: src/naming.adoc [hunk 2]
```diff
@@ -99,10 +99,14 @@ separated from other multi-letter extensions by an underscore, e.g.,
...
 name and an optional version number, or by a numeric name with no version number.
-Other standard extensions that extend
-the supervisor-level architecture are prefixed with the letters "Ss",
-followed by an alphabetical name and an optional version number.  Such
-extensions are defined in Volume II.
+
+Other standard extensions that extend the supervisor-level architecture are 
+named by using ``S'' as a prefix, followed by an alphanumeric name. The name 
+must end with an alphabetical character or an optional 
+version number. The second letter from the bottom cannot be a numeric character
```
Comment (aswaterman) [inline src/naming.adoc:106]: "a numeric character" => "numeric"

Comment (aswaterman) [reply src/naming.adoc:106]: "bottom" => "end"

File: src/naming.adoc [hunk 3]
```diff
@@ -132,10 +136,11 @@ listed, they should be ordered alphabetically.
 
 === Non-Standard Extension Names
 
-Non-standard extensions are named using a single "X" followed by an
-alphabetical name and an optional version number. For example,
-"Xhwacha" names the Hwacha vector-fetch ISA extension; "Xhwacha2"
-and "Xhwacha2p0" name version 2.0 of same.
+Non-standard extensions are named by using a single "X" followed by the alphanumeric 
+name. The name must end with an alphabetic character or an optional version number. The 
+second letter from the bottom cannot be a numeric if the last letter is
```
Comment (aswaterman) [inline src/naming.adoc:141]: "a numeric" => "numeric"

Comment (aswaterman) [reply src/naming.adoc:141]: "bottom" => "end"

**Review by aswaterman — APPROVED** [review]
Feel free to merge after addressing my comments.

File: src/naming.adoc (hunk 1 above)
Comment (kersten1) [inline src/naming.adoc:81]:
```suggestion
optional version number. The second letter from the end cannot be numeric if the
```

File: src/naming.adoc (hunk 2 above)
Comment (kersten1) [inline src/naming.adoc:106]:
```suggestion
version number. The second letter from the end cannot be numeric character
```

File: src/naming.adoc [hunk 4]
```diff
@@ -78,7 +78,7 @@ of the P extension.
 
 Standard unprivileged extensions can also be named by using a single "Z" followed by an
 alphanumeric name. The name must end with an alphabetical character or an
```
Comment (topperc) [inline src/naming.adoc:80]: I think this paragraph is combining two things together. The sentence "The name must end with an alphabetical character or an optional version number" is making version number part of the "name". But, as the "Version Numbers" section says, the version number is encoded _after_ the name. The "name" of extensions must end in alphabetical character independent of whether there is a version number.

I think the "Version Numbering" and "Underscores" sections should be moved further down after the extension naming schemes are described. And all references to versions from the naming sections should be removed.

I'm happy to work on the change.

Comment (aswaterman) [reply src/naming.adoc:80]: @kersten1 I'll be unavailable for a few days, but I'm happy to roll with whatever changes you and Craig agree upon.  Don't wait for me to merge.

File: src/naming.adoc [hunk 5]
```diff
@@ -76,11 +76,11 @@ of the P extension.
...
 
-Standard unprivileged extensions can also be named using a single "Z" followed by
-an alphabetical name and an optional version number. For example,
-"Zifencei" names the instruction-fetch fence extension described in
-<<zifencei>>; "Zifencei2" and
-"Zifencei2p0" name version 2.0 of same.
+Standard unprivileged extensions can also be named by using a single "Z" followed by an
+alphanumeric name. The name must end with an alphabetical character or an 
+optional version number. The second letter from the end cannot be numeric if the
+last letter is ``p''. For example, "Zifencei" names the instruction-fetch fence extension
```
Comment (topperc) [inline src/naming.adoc:82]: I'm not well versed in asciidoc. Does ``p'' mean something? Documentation I found shows ``p`` would be monospace, but I couldn't find anything for mixing `` and ''.

File: src/naming.adoc [hunk 6]
```diff
@@ -144,7 +126,35 @@ For example, an ISA with non-standard extensions Argle and
 Bargle may be named "RV64IZifencei_Xargle_Xbargle".
 
 If multiple non-standard extensions are listed, they should be ordered
-alphabetically.
+alphabetically. Like other multi-letter extensions, they should should be
```
Comment (kersten1) [inline src/naming.adoc:129]:
```suggestion
alphabetically. Like other multi-letter extensions, they should be
```

File: src/naming.adoc [hunk 7]
```diff
@@ -121,21 +101,23 @@ described in the previous section.
...
 
-Non-standard extensions are named using a single "X" followed by an
-alphabetical name and an optional version number. For example,
-"Xhwacha" names the Hwacha vector-fetch ISA extension; "Xhwacha2"
-and "Xhwacha2p0" name version 2.0 of same.
+Non-standard extensions are named by using a single "X" followed by the alphanumeric 
+name. The name must end with an alphabetic character. The 
+second letter from the end cannot be numeric if the last letter is 
+"p". For example, "Xhwacha" names the Hwacha vector-fetch ISA 
+extension;
```
Comment (kersten1) [inline src/naming.adoc:120]:
```suggestion
extension.
```

File: src/naming.adoc (hunk 5 above)
Comment (kersten1) [reply src/naming.adoc]: I'm not sure what is up with that either! Glad you found and fixed it.

**Comment by kersten1** [comment]
@topperc Thanks! Waiting for the checks to complete and I'll merge.

**Pull request merged**